### PR TITLE
Basic string literal support

### LINKED ISSUE #24: Basic string support
Cesium should be able to compile programs with simple string literals (no `wchar_t` or fancy `u8` prefixes, yet; just the basic stuff).

Simple character strings should be compiled down to pinned byte arrays encoded with UTF-8.

### LINKED ISSUE #25: Const support
C programming language includes a complex `const` mechanism to declare objects as `const`. Cesium should support that.

In the nearest future, we'll probably only get an ability to ignore the `const` modifier (because the standard library declares certain arguments as `const`, and I don't want to drop any pieces from the standard library definitions), but eventually we should get the full `const` analysis.

### PR DESCRIPTION
**Pull request by ForNeVeR**
We do the same trick as Roslyn does for the readonly byte arrays.

1. For each literal length, we emit a special `<ConstantPoolItem{length}>` type of desired size.
2. We emit the `<ConstantPool>` type containing a field for each separate string constant, which is being initialized by a data constant.
3. When necessary, we get pointers to these static fields (which are static and thus are safely addressable via pointers).

Closes #24.
Closes #25.

### CODE REVIEW

File: Cesium.Parser.Tests/ParserTests/FullParserTests.cs [hunk 1]
```diff
@@ -67,4 +67,16 @@ int main()
...
 }");
+
+    [Fact]
+    public Task StringLiteral() => DoTest(@"int main()
+{
+    test(""hello world"");
+}");
+
+    [Fact]
+    public Task StringVariable() => DoTest(@"int main()
```
Comment (kant2002) [inline Cesium.Parser.Tests/ParserTests/FullParserTests.cs:78]: Does current implementation handle case of non-const `char*`?

File: Cesium.Parser/CParser.cs [hunk 2]
```diff
@@ -158,13 +161,19 @@ public partial class CParser
...
-        rest?.Insert(0, typeSpecifier) ?? ImmutableArray.Create((DeclarationSpecifier)typeSpecifier);
+        rest?.Insert(0, typeSpecifier) ?? ImmutableArray.Create<IDeclarationSpecifier>(typeSpecifier);
+
+    [Rule("declaration_specifiers: type_qualifier declaration_specifiers?")]
+    private static DeclarationSpecifiers MakeDeclarationSpecifiers(
+        TypeQualifier typeQualifier,
+        DeclarationSpecifiers? rest) =>
+        rest?.Insert(0, typeQualifier) ?? ImmutableArray.Create<IDeclarationSpecifier>(typeQualifier);
+
+    // TODO: [Rule("declaration_specifiers: function_specifier declaration_specifiers?")]
```
Comment (kant2002) [inline Cesium.Parser/CParser.cs:175]: Shouldn't these TODO located near `MakeDeclarationSpecifiers`?

**Review by kant2002 — COMMENTED** [review]
Has couple questions

File: Cesium.Parser.Tests/ParserTests/FullParserTests.cs (hunk 1 above)
Comment (ForNeVeR) [reply Cesium.Parser.Tests/ParserTests/FullParserTests.cs:78]: As of now, `const` is read and then almost completely ignored by the compiler.

(the only exception is that we'll throw an error on `const const` declaration; that's it)

Also, I've tried reading the standard yesterday, but failed to figure out if it's even allowed to have a non-`const` string literal. Is it?

File: Cesium.Parser/CParser.cs (hunk 2 above)
Comment (ForNeVeR) [reply Cesium.Parser/CParser.cs:175]: The order of these TODOs strictly corresponds to the standard, for easier orientation.

**Pull request merged**